src/services/chain/apis/extrinsic.ts

```diff
@@ -116,4 +119,67 @@ export const createMsaForProvider = async (callback: DsnpCallback,
         .catch((error: any) => {
             errorCallback(error);
         });
+}
+
+export const fetchAllSchemas = async (): Promise<Array<SchemaDetails>> => {
+  const api = requireGetProviderApi();
+
+  const schema_id = await api.rpc.schemas.getLatestSchemaId();
+
+  let returnList: Array<SchemaDetails> = [];
+  for (let i = 1; i <= schema_id; i++) {
+    try {
+      let s = await fetchSchema(i);
+      returnList.push(s);
+    } catch (e) {
+      console.log("Error when fetching schemas: {}", e);
+      continue;
+    }
+  }
+  return returnList;
+};
+
+export const fetchSchema = async (schemaId: number): Promise<SchemaDetails> => {
+  const api = requireGetProviderApi();
+
+  const schema = await api.rpc.schemas.getBySchemaId(schemaId);
+  let schemaResult = schema.unwrap();
+  const jsonSchema = Buffer.from(schemaResult.model).toString("utf8");
+  const modelParsed = JSON.parse(jsonSchema);
+  const { schema_id, model_type, payload_location } = schemaResult;
+  return {
+    key: schema_id.toString(),
+    schema_id: schema_id.toString(),
+    model_type: model_type.toString(),
+    payload_location: payload_location.toString(),
+    model_structure: modelParsed
+    }
+};
```

Comment:
This formatted result is probably not ideal in the extrinsic - but as we add more schemas to the demo, will refactor